cmd/result-aggregator/main_test.go

```diff
@@ -148,3 +149,72 @@ func TestLoginHandler(t *testing.T) {
 		}
 	}
 }
+
+func TestValidatePodScalerRequest(t *testing.T) {
+	var testCases = []struct {
+		name     string
+		request  *podScalerRequest
+		expected error
+	}{
+		{
+			name: "everything ok",
+			request: &podScalerRequest{
+				workloadName:     "name",
+				configuredMemory: "100",
+				determinedMemory: "400",
+			},
+			expected: nil,
+		},
+		{
+			name: "empty workload name",
+			request: &podScalerRequest{
+				workloadName:     "",
+				configuredMemory: "100",
+				determinedMemory: "400",
+			},
+			expected: fmt.Errorf("workload_name field in request is empty"),
+		},
+		{
+			name: "empty configured memory",
+			request: &podScalerRequest{
+				workloadName:     "name",
+				configuredMemory: "",
+				determinedMemory: "400",
+			},
+			expected: fmt.Errorf("configured_memory field in request is empty"),
+		},
+		{
+			name: "empty determined memory",
+			request: &podScalerRequest{
+				workloadName:     "name",
+				configuredMemory: "100",
+				determinedMemory: "",
+			},
+			expected: fmt.Errorf("determined_memory field in request is empty"),
+		},
+		{
+			name: "empty determined memory",
+			request: &podScalerRequest{
+				workloadName:     "name",
+				configuredMemory: "100",
+				determinedMemory: "",
+			},
+			expected: fmt.Errorf("determined_memory field in request is empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			actual := validatePodScalerRequest(testCase.request)
+			if actual == nil {
+				if testCase.expected != nil {
+					t.Fatalf("got error when test case was ok, error: %v", actual.Error())
+				}
+			} else {
+				if actual.Error() != testCase.expected.Error() {
+					t.Fatalf("error: %v doesn't match expected error: %v", actual.Error(), testCase.expected.Error())
+				}
```

Comment:
You can simplify this by using `cmp.Diff`. See an example here: https://github.com/openshift/ci-tools/blob/6aa22d5527e30fa0c99e19b11064ffb4e94b2e1b/cmd/ci-operator/main_test.go#L1150-L1152